apm-collector/apm-collector-worker/src/main/java/org/skywalking/apm/collector/worker/httpserver/AbstractPost.java

```diff
@@ -40,35 +42,66 @@ final public void onWork(Object message) throws Exception {
             this.ownerWorkerRef = ownerWorkerRef;
         }
 
-        @Override
-        final protected void doPost(HttpServletRequest request,
-                                    HttpServletResponse response) throws ServletException, IOException {
+        /**
+         * Get segment's buffer from request then execute deserialize operation.
+         *
+         * @param request an {@link HttpServletRequest} object that contains the request the client has made of the
+         * servlet
+         * @param response {@link HttpServletResponse} object that contains the response the servlet sends to the
+         * client
+         * @throws ServletException if the request for the POST could not be handled
+         * @throws IOException if an input or output error is detected when the servlet handles the request
+         */
+        @Override final protected void doPost(HttpServletRequest request,
+            HttpServletResponse response) throws ServletException, IOException {
             JsonObject resJson = new JsonObject();
             try {
                 BufferedReader bufferedReader = request.getReader();
                 streamReader(bufferedReader);
                 reply(response, resJson, HttpServletResponse.SC_OK);
             } catch (Exception e) {
-                logger.error(e);
+                logger.error(e, e);
                 resJson.addProperty("error", e.getMessage());
                 reply(response, resJson, HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
             }
         }
 
+        /**
+         * Read segment's buffer from buffer reader by stream mode. when finish read one segment then send to analysis.
+         * This method in there, so post servlet just can receive segments data.
```

Comment:
Read segments from buffer in streaming mode. Send to analysis after each one of segments deserialized. 

I can't understand about `This method in there, so post servlet just can receive segments data.`.